lib/spack/llnl/util/filesystem.py

```diff
@@ -1311,29 +1345,34 @@ def traverse_tree(
 
 def lexists_islink_isdir(path):
     """Computes the tuple (lexists(path), islink(path), isdir(path)) in a minimal
-    number of stat calls."""
-    # First try to lstat, so we know if it's a link or not.
-    try:
-        lst = os.lstat(path)
-    except (IOError, OSError):
-        return False, False, False
+    number of stat calls on unix. Use os.path and symlink.islink methods for windows."""
+    if sys.platform == "win32":
+        if not os.path.lexists(path):
+            return False, False, False
+        return os.path.lexists(path), islink(path), os.path.isdir(path)
+    else:
+        # First try to lstat, so we know if it's a link or not.
```

Comment:
Can you remove this `else` branch and remove indent, it's unnecessary and only makes the diff large.

Also, can't you define the function differently on windows instead of branching at runtime?

Reply:
This method appears to only be used in tests, and since it is now used on tests that run cross-platform, something has to be decided at runtime based on OS. I don't think there's much overhead here...